src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -3114,6 +3160,42 @@ void MergeTreeData::checkAlterIsPossible(const AlterCommands & commands, Context
             }
         }
 
+        // Technically it is possible to specify the same column for `version` and `is_deleted`, thus let's be sure and don't use -if-else here
```

Comment:
I didn't see a good reason why not to allow apart from I couldn't imagine a situation where it can be useful. If you also don't see a reason to do, then I can add another check to prevent that.